### ISSUE TITLE
Online / Offline indicator

### ISSUE THREAD

**Issue by jlydia** [amidaware/tacticalrmm#1506]
**Is your feature request related to a problem? Please describe.**
You have two indicators for the Agent Status (see attached), but none for what we look for at a quick glance, which is whether the device is online or off.

**Describe the solution you'd like**
It would be great if you would change the "Agent Status" indicator to instead represent Online/Offline status

**Describe alternatives you've considered**
There are no time-saving alternatives as we have to currently click the filter button to get that status.

**Additional context**
See screenshot of the two current indicators for agent status.

![two agent update statuses](https://user-images.githubusercontent.com/43318108/236589461-40886bf7-2d84-46db-9abb-217ab746c6ea.png)

**Comment by silversword411** [comment]
A TRMM agent is sending a NATS "alive" packet every 120 seconds or less. This is updating the TRMM server's "agent status" column. 

You need to use the relevant refresh button (Review https://docs.tacticalrmm.com/tipsntricks/#understanding-refreshing-trmm-data) to update your local web browser with the servers up-to-the-second agent last alive packet.

With that understanding, I think this is an invalid issue and will close it. Post questions if you have another perspective and think there's room for improvement.

**Comment by jlydia** [comment]
I fully
          understand how that process works, but that was not what I was
          referring to.
    Most
          RMMs that I have used have an "Online" indicator for the
          device, such as this one, which lets us see quickly which
          devices are currently online:
    
    TacticalRMM does not have a colume like this but does use two
      columns to alert us about the "Agent Status".
    
    Why not use one of those columns (the "Agent Status" column, at
      the right), to indicate "Online/Offline", this way we can easily
      tell when a device is online or offline.
    -JL
    
    On 5/5/23 9:29 PM, silversword411
      wrote:
    
    
      
      A TRMM agent is sending a NATS "alive" packet every
        120 seconds or less. This is updating the TRMM server's "agent
        status" column.
      You need to use the relevant refresh button (Review
        https://docs.tacticalrmm.com/tipsntricks/#understanding-refreshing-trmm-data)
        to update your local web browser with the servers
        up-to-the-second agent last alive packet.
      With that understanding, I think this is an invalid
        issue and will close it. Post questions if you have another
        perspective and think there's room for improvement.
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        You are receiving this because you authored the thread.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/amidaware/tacticalrmm/issues/1506#issuecomment-1536964703",
"url": "https://github.com/amidaware/tacticalrmm/issues/1506#issuecomment-1536964703",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by bbrendon** [comment]
Hi @jlydia 
The purpose of the "Agent Overdue" (as you have in your original issue) is just that. It is the "Online/Offline" indicator. If its red/orange, then the agent is not online. If it's green, then it is online.

**Comment by silversword411** [comment]
You are incorrect. This:

> Most RMMs that I have used have an "Online" indicator for the device, such as this one, which lets us see quickly which devices are currently online:

is an equivalent of this:

![2023-05-05 22_27_40-Online _ Offline indicator · Issue #1506 · amidaware_tacticalrmm - Brave](https://user-images.githubusercontent.com/13395348/236593956-d8295017-f3bf-4cee-bf4b-a95cb2ce2ed4.png)

The difference between your other RMM and TRMM....is you can control the times it changes from green, to yellow, to red down to a 120 second time scale with

![2023-05-05 21_30_06-agent offline overdue](https://user-images.githubusercontent.com/13395348/236594003-69f94071-5908-4436-8a70-4b044b569a26.png)

**Comment by jlydia** [comment]
My
          apology for not explaining this clear enough.
    I am not
          referring to the "Agent" online status, we all understand how
          this works in TRMM.
    I am
          referring to a "Device" online status indicator.
    I would
          be great if we had one and since you have two for the "Agent"
          why not use one of these to indicate the "Device" online
          status.
    Again,
          this is not about the "Agent" status, but the "Device" status.
        
    -JL
    
    On 5/5/23 10:31 PM, silversword411
      wrote:
    
    
      
      You are incorrect. This:
      
        Most RMMs that I have used have an "Online"
          indicator for the device, such as this one, which lets us see
          quickly which devices are currently online:
      
      is an equivalent of this:
      
      The difference between your other RMM and TRMM....is
        you can control the times it changes from green, to yellow, to
        red down to a 120 second time scale with
      
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        You are receiving this because you were mentioned.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/amidaware/tacticalrmm/issues/1506#issuecomment-1536990286",
"url": "https://github.com/amidaware/tacticalrmm/issues/1506#issuecomment-1536990286",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by silversword411** [comment]
> Again, this is not about the "Agent" status, but the "Device" status.

The only thing TRMM can tell you is the status of the TRMM (RMM) "agent" running on the "device", because that is the only way any RMM server knows anything about a "device". Same as your other RMM.

Unless you can tell me your other RMM, and the technology it's "status" indicator uses to determine a devices online/offline color I can't help describe to you with any further detail your misunderstanding on what an agent red vs green status indicator means.

**Comment by jlydia** [comment]
My other RMM is Datto, but I have no idea what they use to indicate whether the device is online or not.

...and "yes" I understand how the Agent status works from green to yellow to red based on the settings.
--
If you are saying TRMM's Agent status = Device status, then why are some devices with "Agent" status = red, but the devices are actually Online?

![image](https://user-images.githubusercontent.com/43318108/236596772-75d93efe-4067-4169-8d65-5ea0cca65a3d.png)

**Comment by bbrendon** [comment]
Can this be converted to a discussion?

You're talking about mesh now. Not tactical.  I'd say read the section on meshcentral.
https://docs.tacticalrmm.com/howitallworks/

**Comment by silversword411** [comment]
> Can this be converted to a discussion?

Nope, so now we're actually having a detailed technical discussion on the technologies and I know what @jlydia is talking about.

So you're talking about the discrepancy of sensing the device/agent status based on two different technologies.

- MeshCentral Agent: Uses websockets which as best as I've been able to tell is on a 15-30 second timescale at sensing a "connected/disconnected" status (there's a youtube video from Ylian discussing websockets, you should find/watch)
- TRMM agent: Uses NATS keepalive packets that are randomly triggered in the 90-120 second timescale and are updating the proverbial last seen date/time in the django/postgres database, and requires the refresh button to be clicked to reload the status from the database.

So this enhancement request is wanting the TRMM agent to use a native websocket connection to indicate an agents online/offline status instead of NATS. I've actually talked with wh1te about this, and long-term I believe he may be considering re-engineering TRMM to go websocket for more connect information details...but it's a major re-engineering rework. So I believe this is actually on a long-term TODO list.

I'll let wh1te chime in....and possibly re-open this (and update the topic to something like "Switch TRMM agent online status indicator from NATS to websocket for status indicator") depending on where he is on a if/when this might be reworked.

**Comment by jlydia** [comment]
> So you're talking about the discrepancy of sensing the device/agent status based on two different technologies.
> 
> * MeshCentral Agent: Uses websockets which as best as I've been able to tell is on a 15-30 second timescale at sensing a "connected/disconnected" status (there's a youtube video from Ylian discussing websockets, you should find/watch)
> * TRMM agent: Uses NATS keepalive packets that are randomly triggered in the 45-90 second timescale and are updating the proverbial last seen date/time in the django/postgres database, and requires the refresh button to be clicked to reload the status from the database and updated via the vue html admin interface.
> 
> So this enhancement request is wanting the TRMM agent to use a native websocket connection to indicate an agents online/offline status instead of NATS. ...
> 

Hey @silversword411 , you detailed exactly what I was attempting to convey, but struggled to articulate. I could only throw out pics. I had forgotten that the "Take Control" was MeshCentral. Thanks for the reminder and again, "yes" this is what I was questioning. Would be nice is we could "Switch TRMM agent online status indicator from NATS to websocket for agent status".

**Comment by dinger1986** [comment]
If agents are showing off in trmm but on in mesh, sometimes it's because the trmm agent hasn't been updated or otherwise has gone offline. What version is the agent?

**Comment by wh1te909** [comment]
Reopened cuz yea i do eventually want to switch to a realtime websocket for agent status similar to how mesh does it

**Comment by bsnipes** [comment]
I just want to add a bit of info for my system that might be what the OP was seeing. After updating Tactical about a week ago, all of my agents appeared with red status even though I could remote them. I had to run an "./update --force" to get it showing the proper status. No clue exactly what happened but a reboot of the server didn't have any effect on showing the correct agent status. I originally thought reinstalling the Tactical agent or Mesh agent would fix the issue but it didn't which is why I did the force update.

**Comment by dinger1986** [comment]
you sure there wasnt an issue with certs or something stopping nats working?

**Comment by bsnipes** [comment]
> you sure there wasnt an issue with certs or something stopping nats working?

Cert is from 4/7 through 7/6. I tried re-installing clients, a server reboot, and a standard update prior to performing the --force option.
